.travis.yml

```diff
@@ -1,15 +1,45 @@
-services:
-  - docker
-
 dist: xenial
 
-before_install:
-  - docker pull getsentry/snuba:latest || true
-  - docker build -t getsentry/snuba . --cache-from getsentry/snuba:latest
-  - docker network create --attachable cloudbuild
+branches:
+  only:
+    - master
+
+cache:
+  pip: true
+  npm: true
+
+stages: [pre_commit, test]
 
-script:
-  - SNUBA_IMAGE=getsentry/snuba docker-compose -f docker-compose.gcb.yml run --rm snuba-test
+matrix:
+  fast_finish: true
+  include:
+    - name: 'pre-commit hooks (includes python formatting + linting)'
+      stage: pre_commit
+      language: python
+      python: 3.7
+      install:
+        - make setup-git
+      script:
+        # Run pre-commit to lint and format check files that were changed (but not deleted) compared to master.
+        # XXX: there is a very small chance that it'll expand to exceed Linux's limits
+        #      `getconf ARG_MAX` - max # bytes of args + environ for exec()
+        - pre-commit run --files $(git diff --diff-filter=d --name-only master)
```

Comment:
I kind of wonder if this should also special case when `test $(git rev-parse head) = $(git rev-parse master)` to check all files in case somehow something sneaks in somehow that breaks master (or the lint config changes, etc.), and it's not caught until somebody touches the file.

Reply:
Oh, that's a good idea. Then branch to `pre-commit run --all-files` I'm assuming.

It'd make master builds a bit slower, but not by much here. I can follow up with that in the future, let's see if master drifts at all in a month.

Reply:
(I'd bet it won't.)